### ISSUE TITLE
FEATURE REQUEST: "Microsoft Docs Alerts" style Callouts

### ISSUE THREAD

**Issue by shu307** [oprypin/markdown-callouts#13]
[Microsoft Docs Alerts](https://learn.microsoft.com/en-us/contribute/markdown-reference#alerts-note-tip-important-caution-warning)
Supported by Obsidian: [obsidian callouts](https://help.obsidian.md/How+to/Use+callouts)

**Comment by oprypin** [comment]
You can fork the extension and just change the regular expression, should be relatively easy. Let me know if you run into trouble. This syntax doesn't match the goals of the extension, so will not be supported

**Comment by oprypin** [comment]
https://oprypin.github.io/markdown-callouts/#project-goals

**Comment by oprypin** [comment]
Hmm well maybe 😃

**Comment by oprypin** [comment]
@facelessuser  @llucax 

The background on what "GitHub style" meant historically is here: https://github.com/oprypin/markdown-callouts/issues/16#issuecomment-1817494201
This extension was made before it became a thing but I've been following the topic with interest.

So I'd be interested in also supporting the "obsidian" style which is now the "GitHub" style.
However this style would be very limited, as they don't define any ability to set a custom title of the admonition, or to set different CSS classes. The syntax is very simple and very "unforgiving" - nothing extra is allowed on the first line.
So if the extra features *are* supported, it would no longer provide graceful degradation on GitHub, which is not aligned with this project's goals but I keep thinking about it in the background.

Base example (really the only thing that works on GitHub):

> [!NOTE]
> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

---


</details>

There is this project which simply moved forward and made some additional syntaxes. (Also started before GitHub admonitions were a thing)
https://github.com/sondregronas/mkdocs-callouts

Which as you can see doesn't display as anything good on GitHub.

<details><summary>Custom title example (still Obsidian-compliant but not GitHub-compliant):</summary>

> [!NOTE] Custom title
> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

---


</details>

<details><summary>Foldable block example (still Obsidian-compliant but not GitHub-compliant):</summary>

> [!NOTE]-
> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

---

</details>

<details><summary>Custom class example (totally made up by that plugin):</summary>

> [!NOTE inline end]
> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

---

</details>

However I am not happy with just *only sometimes* having graceful degradation, or with how that project is implemented (it's a simplistic find&replace rather than a Markdown extension)

So, interested in making *something* but unsure how to proceed.

Actually if our goal is GitHub only, then we could start inventing yet another syntax, such as:

> [!NOTE]
> **Custom title.**
> Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

But can't think of anything remotely reasonable for custom classes or foldable blocks.

**Comment by oprypin** [comment]
And  @llucax would you really be happy with the pure GitHub admonition style considering it supports only the most basic type of admonition with 0 extra features? What would you do when you decide that you want a custom title?
This is why I believe that this extension's current style may still be the most flexible solution.

**Comment by facelessuser** [comment]
Understood. My interest is mainly to understand the direction of this project for the sake of redirects. There are a lot of people who orbit around the GitHub "way", and they often like things to be consistent, even if that means limited. This may not be your project's goal, I'm just looking for clarification.

**Comment by llucax** [comment]
> And @llucax would you really be happy with the pure GitHub admonition style considering it supports only the most basic type of admonition with 0 extra features?

Yes. For me it adds value. My main use case is having a README.md that can be seen nicely in GitHub but also can be included in mkdocs and also seen nicely there. My goal is not to be able to view all the mkdocs-specific markdown files nicely in GitHub.

> What would you do when you decide that you want a custom title?

Suck it up and not have it. For my restricted use case, is not that important to have custom titles.

This is how the README of the project with more thought about documentation we have looks like:

https://github.com/frequenz-floss/frequenz-channels-python/

(I prefer removing the `inline end` if it brings nicer syntax to GitHub). Then the readme is included in the docs, see for example:

* https://github.com/frequenz-floss/frequenz-channels-python/blob/v1.x.x/docs/index.md?plain=1
* https://frequenz-floss.github.io/frequenz-channels-python/v1.0-dev/

**Comment by oprypin** [comment]
https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types
"CAUTION" can be represented as "DANGER" class.
What could we represent "IMPORTANT" as? Actually probably best left in the default style

**Comment by oprypin** [comment]
I went ahead and implemented GitHub-style syntax, matching its behavior as strictly as possible

* https://github.com/oprypin/markdown-callouts/pull/18

Try it out and let me know if you can find any mismatching edge cases :D

```bash
pip install git+https://github.com/oprypin/markdown-callouts.git@refs/pull/18/head
```

**Comment by oprypin** [comment]
and the other part of instructions is
```yaml
markdown_extensions:
  - github-callouts
```

**Comment by llucax** [comment]
```
ERROR   -  Config value 'markdown_extensions': Failed to load extension 'github-callouts'.
           ModuleNotFoundError: No module named 'github-callouts'
```

:-/

**Comment by llucax** [comment]
> [squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types) "CAUTION" can be represented as "DANGER" class. What could we represent "IMPORTANT" as? Actually probably best left in the default style

Right. Yeah. I guess that can be left to the user to properly customize, or have a separate feature request to add caution and important admonitions to `mkdocs-material`. I actually find myself several times wanting *important*, and was going to add a custom admonition for it anyways.

**Comment by llucax** [comment]
> ```yaml
> markdown_extensions:
>   - github-callouts
> ```

Also silly question, any reason not to call the extension `github-alerts` instead, as this is the official GitHub name for them?

**Comment by oprypin** [comment]
I merged and released this.

* https://github.com/oprypin/markdown-callouts/pull/18

The only documentation is release notes, didn't write anything else yet.
https://github.com/oprypin/markdown-callouts/releases/tag/v0.4.0
```bash
pip install -U markdown-callouts
```

**Comment by jayqi** [comment]
This is a godsend of a feature, and IMO it should be loudly featured in the project README.

**Comment by hbgodseeker** [comment]
Glad to see the support for obsidian callouts

> And @llucax would you really be happy with the pure GitHub admonition style considering it supports only the most basic type of admonition with 0 extra features? What would you do when you decide that you want a custom title? This is why I believe that this extension's current style may still be the most flexible solution.

About the custom title problem you mentioned before
I found that [obsidian callouts](https://help.obsidian.md/Editing+and+formatting/Callouts) has carried out a solution like below

![Change the title](https://github.com/user-attachments/assets/20ceab11-6611-4af3-a174-462a807a3e76)

Can you add this feature to the project? This will really improve the experience of using Obsidian+mkdocs to publish my articles. @oprypin
